src/map/utils/zoneutils.h

```diff
@@ -43,17 +43,19 @@ namespace zoneutils
 
     int GetWeatherElement(WEATHER weather);
 
-    CZone*       GetZone(uint16 ZoneID);
-    CNpcEntity*  GetTrigger(uint16 TargID, uint16 ZoneID);
-    CBaseEntity* GetEntity(uint32 ID, uint8 filter = -1);
-    CCharEntity* GetCharByName(std::string const& name);
-    CCharEntity* GetCharFromWorld(uint32 charid, uint16 targid);  // returns pointer to character by id and target id
-    CCharEntity* GetChar(uint32 id);                              // returns pointer to character by id
-    CCharEntity* GetCharToUpdate(uint32 primary, uint32 ternary); // returns pointer to preferred char to update for party changes
-    void         ForEachZone(const std::function<void(CZone*)>& func);
-    uint64       GetZoneIPP(uint16 zoneid);                      // returns IPP for zone ID
-    bool         IsResidentialArea(CCharEntity*);                // returns whether or not the area is a residential zone
-    bool         IsAlwaysOutOfNationControl(REGION_TYPE region); // returns true if a region should never trigger "in areas outside own nation's control" latent effect; false otherwise.
+    CZone*              GetZone(uint16 ZoneID);
+    CNpcEntity*         GetTrigger(uint16 TargID, uint16 ZoneID);
+    CBaseEntity*        GetEntity(uint32 ID, uint8 filter = -1);
+    CCharEntity*        GetCharByName(std::string const& name);
+    CCharEntity*        GetCharFromWorld(uint32 charid, uint16 targid);  // returns pointer to character by id and target id
+    CCharEntity*        GetChar(uint32 id);                              // returns pointer to character by id
+    CCharEntity*        GetCharToUpdate(uint32 primary, uint32 ternary); // returns pointer to preferred char to update for party changes
+    std::vector<uint16> GetZonesOnThisProcess();
```

Comment:
This can be auto and trailing return so you don't have to move all the other funcitons